improver_tests/psychrometric_calculations/hail_size/test_hail_size.py

```diff
@@ -0,0 +1,314 @@
+    of the produced hail_size cube"""
+    ccl_pressure.data = np.full_like(ccl_pressure.data, ccl_p)
+    ccl_temperature.data = np.full_like(ccl_temperature.data, ccl_t)
+    relative_humidity_on_pressure.data = np.full_like(
+        relative_humidity_on_pressure.data, humidity
+    )
+
+    result = HailSize()(
+        ccl_temperature,
+        ccl_pressure,
+        temperature_on_pressure_levels,
+        relative_humidity_on_pressure,
+    )
+
+    np.testing.assert_array_equal(result.data, expected)
+    metadata_check(result)
+
+
+def test_temperature_too_high(
+    temperature_on_pressure_levels,
+    ccl_pressure,
+    ccl_temperature,
+    relative_humidity_on_pressure,
+):
+    """Tests for the case where there are grid squares where the temperature
+    doesn't drop below 268.15K at any pressure."""
+    temperature_on_pressure_levels.data = np.full_like(
+        temperature_on_pressure_levels.data, 260
+    )
+    temperature_on_pressure_levels.data[:, :, 1] = 300
+
+    expected = [[[35, 35], [0, 0], [35, 35]], [[35, 35], [0, 0], [35, 35]]]
+
+    result = HailSize()(
+        ccl_temperature,
+        ccl_pressure,
+        temperature_on_pressure_levels,
+        relative_humidity_on_pressure,
+    )
+    np.testing.assert_array_equal(result.data, expected)
+    metadata_check(result)
+
+
+@pytest.mark.parametrize(
+    "variable",
+    (
+        "temperature_on_pressure_levels",
+        "ccl_temperature",
+        "ccl_pressure",
+        "relative_humidity_on_pressure",
+    ),
+)
+def test_spatial_coord_mismatch(variable, request):
+    """Tests that an error is raised if the spatial
+    coordinates of all cubes don't match"""
+
+    variable_new = request.getfixturevalue(variable)
+    variable_slice = next(variable_new.slices_over("longitude"))
+
+    fixtures = [
+        "ccl_temperature",
+        "ccl_pressure",
+        "temperature_on_pressure_levels",
+        "relative_humidity_on_pressure",
+    ]
+    fixtures.remove(variable)
+    cubes = CubeList(request.getfixturevalue(fix) for fix in fixtures)
+    cubes.append(variable_slice)
+
+    ccl_temperature = cubes.extract("temperature_at_cloud_condensation_level")
+    ccl_pressure = cubes.extract("pressure_at_cloud_condensation_level")
+    temperature_on_pressure = cubes.extract("temperature_on_pressure_levels")
+    relative_humidity_on_pressure = cubes.extract(
+        "relative_humidity_on_pressure_levels"
+    )
+
+    with pytest.raises(ValueError):
+        HailSize()(
+            ccl_temperature[0],
+            ccl_pressure[0],
+            temperature_on_pressure[0],
+            relative_humidity_on_pressure[0],
+        )
+
+
+@pytest.mark.parametrize(
+    "variable",
+    (
+        "temperature_on_pressure_levels",
+        "ccl_temperature",
+        "ccl_pressure",
+        "relative_humidity_on_pressure",
+    ),
+)
+def test_incorrect_units(variable, request):
```

Comment:
This is a unit test for `cf_units` rather than the code you have written. In general, unless we are modifying the returned error message, or handling it in an interesting way, we do not need to test the exceptions raised by other libraries that we are using. I would delete this test.

Reply:
Good to know. Removed the test